go/worker/storage/committee/node.go

```diff
@@ -838,6 +849,12 @@ func (n *Node) worker() { // nolint: gocyclo
 		}
 	}
 
+	// Notify the checkpointer of the genesis round so it can be checkpointed.
+	if n.checkpointer != nil {
+		n.checkpointer.NotifyNewVersion(genesisBlock.Header.Round)
+		n.checkpointer.Flush()
```

Comment:
Could potentially also implement "flushing" within the `NotifyNewVersion` method on the checkpointer itself, and not expose it via a separate method. Unless there is a reason not to flush always when notifying of a new version.

Reply:
Yeah the reason is to not do the checkpoint checks on each received block, but rather to batch the checks using a configurable timeout (because there is no sense to check this too often).

Reply:
makes sense :+1: